gcsfs/core.py

```diff
@@ -1347,15 +1322,10 @@ def validate_response(self, status, content, json, path, headers=None):
             else:
                 raise RuntimeError(msg)
         else:
-            if headers is not None and "X-Goog-Hash" in headers:
-                # if header includes md5 hash, check that data matches
-                bits = headers["X-Goog-Hash"].split(",")
-                for bit in bits:
-                    key, val = bit.split("=", 1)
-                    if key == "md5":
-                        md = b64decode(val)
-                        if md5(content).digest() != md:
-                            raise ChecksumError("Checksum failure")
+            if headers is not None:
```

Comment:
Should this be optional, can md5 be missing here?

Reply:
Some of the tests pass in `headers=None` so I tried to keep the logic similar:
https://github.com/VulcanClimateModeling/gcsfs/blob/c76f93ce1e9bae5ba05fbbd1833d1608495444db/gcsfs/tests/test_core.py#L951-L951

I'm not sure if this can happen with real GCS responses...probably not.

Reply:
I'm making the if-statement identical to the current version to ensure backwards compatibility.

Reply:
Yes, but I mean that we explicitly use MD5 here, which may be unnecessary and could perhaps be missing, like it was for your case. Or maybe in this case it is indeed guaranteed to be there, I don't know.

Reply:
I see your point. I didn't modify this since the "consistency" flag was not used here. I believe this check is only needed for `_cat_file` since `_{get,put,pipe}_file` have their own validation logic using `consistency`. The checking logic is unique to each operation, so would be hard to implement in `_call`.

I'd propose:
- adding validation to `_cat_file`
- removing the checksum logic from `validate_response/_call`.

Reply:
I could do this in a follow-up PR if you think this is otherwise ready to merge @martindurant.

Reply:
Actually let's leave it until someone needs this. If ever there is a similar exception to the one you saw, the user would naturally find their way here.